controller/lib/src/etcd/mod.rs

```diff
@@ -0,0 +1,127 @@
+use etcd_client::{Client, DeleteResponse, Error, GetOptions, PutResponse};
+use log::info;
+
+pub struct EtcdClient {
+    inner: Client,
+}
+
+impl EtcdClient {
+    pub async fn new(address: String) -> Self {
+        info!("Starting ETCD client on {}", address);
+        EtcdClient {
+            inner: Client::connect([address], None).await.unwrap(),
+        }
+    }
+
+    pub async fn get(&mut self, key: &str) -> Result<String, Error> {
```

Comment:
Following the same logic as with `delete`, I don't think not finding a value for a given key is an error. If there's no kv for a given k, you return `None`. That's kind of what the `Option` type is for.

So your get could simply be:

```rust
self.inner.get(key, None).await.unwrap().kvs().first()
```

(without the unwrap)